### ISSUE TITLE
Check if getting sims 2 times is needed

### ISSUE THREAD

**Issue by mihaiblaga89** [flexn-io/renative#988]
```
[ task ] [run] getAppleDevices[1] ignoreDevices:true ignoreSimulatorsfalse
✔ Executing: xcrun xctrace list devices
✔ Executing: xcrun simctl list --json
[ task ] [run] launchAppleSimulator[1] iPhone 8
[ task ] [run] getAppleDevices[2] ignoreDevices:true ignoreSimulatorsundefined
✔ Executing: xcrun xctrace list devices
✔ Executing: xcrun simctl list --json
```

**Comment by GabrieleKaceviciute** [comment]
@mihaiblaga89 I can't reproduce it on 1.0.0-rc.18
```
➜  app-harness git:(1.0.0-rc.18) npx rnv run -p ios

┌─────────────────────────────────────────────────────────────────┐
│ ██████╗ ███████╗███╗   ██╗ █████╗ ████████╗██╗██╗   ██╗███████╗ │
│ ██╔══██╗██╔════╝████╗  ██║██╔══██╗╚══██╔══╝██║██║   ██║██╔════╝ │
│ ██████╔╝█████╗  ██╔██╗ ██║███████║   ██║   ██║██║   ██║█████╗   │
│ ██╔══██╗██╔══╝  ██║╚██╗██║██╔══██║   ██║   ██║╚██╗ ██╔╝██╔══╝   │
│ ██║  ██║███████╗██║ ╚████║██║  ██║   ██║   ██║ ╚████╔╝ ███████╗ │
│ ╚═╝  ╚═╝╚══════╝╚═╝  ╚═══╝╚═╝  ╚═╝   ╚═╝   ╚═╝  ╚═══╝  ╚══════╝ │
│ 🚀 v:1.0.0-rc.18 | renative.org | 5/9/2024, 9:54:54 AM          │
│ $ rnv run -p ios                                                │
└─────────────────────────────────────────────────────────────────┘
info: Current engine: @rnv/engine-rn (../engine-rn)
task: ○ run ↺1
task: ○ configure ↺1
task: ○ platform configure ↺1
task: ○ project configure ↺1
task: ○ workspace configure ↺1
task: ✔ workspace configure
task: ○ app configure ↺1
info: Current app config: harness
task: ✔ app configure
info: ../../node_modules/@react-native-community/cli-platform-ios/build/tools/listIOSDevices.js overriden by: /@rnv/config-templates/pluginTemplates/@react-native-community/cli-platform-ios/overrides@11.3.7.json
info: ../../node_modules/@react-native-community/cli/build/bin.js overriden by: /@rnv/config-templates/pluginTemplates/@react-native-community/cli/overrides.json
warn: ○ project configure: No Match found in /react-native-tvos/Libraries/Components/ScrollView/ScrollViewNativeComponent.js for expression: NativeComponentRegistry.get<Props>(
    'RCTScrollView'. source: /@rnv/config-templates/pluginTemplates/react-native-tvos/overrides.json
warn: ○ project configure: No Match found in /react-native-tvos/Libraries/Components/ScrollView/ScrollViewNativeComponent.js for expression: Platform.OS === 'android'
    ? {
        uiViewClassName: 'RCTScrollView'. source: /@rnv/config-templates/pluginTemplates/react-native-tvos/overrides.json
info: ../../node_modules/react-native-photo-editor overriden by: /@rnv/config-templates/pluginTemplates/react-native-photo-editor/overrides
task: ✔ project configure
info: Current Build Scheme: debug
task: ✔ platform configure
info: Found custom assetSources at ../../node_modules/@flexn/assets-renative-outline/assets/ios. Will be used to generate assets.
info: Podfile.lock does not exist. Will execute pod actions...
✔ Executing: bundle install
✔ Executing: bundle exec pod install
task: ✔ configure
✔ Executing: xcrun simctl list --json
task: ○ start ↺1
task: ○ configureSoft ↺1
task: ○ app configure ↺2
info: Current app config: harness
task: ✔ app configure
info: Current Build Scheme: debug
task: ✔ configureSoft

Dev server running at: http://0.0.0.0:8092/index.bundle?platform=ios

task: ✔ start
⠴ Waiting for bundler to finish...warn Package rn-fetch-blob contains invalid configuration: "dependency.hooks" is not allowed. Please verify it's properly linked using "npx react-native config" command and contact the package maintainers about this.
warn =================================================================================================
warn From React Native 0.73, your project's Metro config should extend '@react-native/metro-config'
warn or it will fail to build. Please copy the template at:
warn https://github.com/facebook/react-native/blob/main/packages/react-native/template/metro.config.js
warn This warning will be removed in future (https://github.com/facebook/metro/issues/1018).
warn =================================================================================================
⠦ Waiting for bundler to finish...info Welcome to React Native v0.73
info Starting dev server on port 8092...

                        ▒▒▓▓▓▓▒▒
                     ▒▓▓▓▒▒░░▒▒▓▓▓▒
                  ▒▓▓▓▓░░░▒▒▒▒░░░▓▓▓▓▒
                 ▓▓▒▒▒▓▓▓▓▓▓▓▓▓▓▓▓▒▒▒▓▓
                 ▓▓░░░░░▒▓▓▓▓▓▓▒░░░░░▓▓
                 ▓▓░░▓▓▒░░░▒▒░░░▒▓▒░░▓▓
                 ▓▓░░▓▓▓▓▓▒▒▒▒▓▓▓▓▒░░▓▓
                 ▓▓░░▓▓▓▓▓▓▓▓▓▓▓▓▓▒░░▓▓
                 ▓▓▒░░▒▒▓▓▓▓▓▓▓▓▒░░░▒▓▓
                  ▒▓▓▓▒░░░▒▓▓▒░░░▒▓▓▓▒
                     ▒▓▓▓▒░░░░▒▓▓▓▒
                        ▒▒▓▓▓▓▒▒


                Welcome to Metro v0.80.9
              Fast - Scalable - Integrated


⠋ Waiting for bundler to finish...info React Native v0.74.1 is now available (your project is running on v0.73.4).
info Changelog: https://github.com/facebook/react-native/releases/tag/v0.74.1
info Diff: https://react-native-community.github.io/upgrade-helper/?from=0.74.1
info For more info, check out "https://reactnative.dev/docs/upgrading?os=macos".
✔ Waiting for bundler to finish...
✔ Executing: npx react-native run-ios --scheme=RNVApp --mode=Debug --no-packager --simulator iPhone\ 15
┌─────────────────────────────────────────────────────────────────┐
│ ⚠ BUNDLER STARTED | 5/9/2024, 9:57:19 AM | rnv@1.0.0-rc.18      │
│ $ rnv run -p ios                                                │
├─────────────────────────────────────────────────────────────────┤
│ Project: @rnv/app-harness@1.0.0-rc.18                           │
│ Workspace: rnv                                                  │
│ Platform (-p): ios                                              │
│ Engine: @rnv/engine-rn                                          │
│ Template: @rnv/template-starter                                 │
│ App Config (-c): harness                                        │
│ Build Scheme (-s): debug                                        │
│ Target (-t): iPhone 15                                          │
│ Supported Platforms: ios, android, androidtv, firetv, androidwear, web, tizen, tvos, webos, macos, tizenwatch, kaios, chromecast, linux
│ Env Info: darwin | arm64 | node v18.19.0                        │
│ Executed Time: 0h:2m:24s:500ms                                  │
│ warn: ○ project configure: No Match found in /react-native-tvos/Libraries/Components/ScrollView/ScrollViewNativeComponent.js for expression: NativeComponentRegistry.get<Props>(
    'RCTScrollView'. source: /@rnv/config-templates/pluginTemplates/react-native-tvos/overrides.json
│ warn: ○ project configure: No Match found in /react-native-tvos/Libraries/Components/ScrollView/ScrollViewNativeComponent.js for expression: Platform.OS === 'android'
    ? {
        uiViewClassName: 'RCTScrollView'. source: /@rnv/config-templates/pluginTemplates/react-native-tvos/overrides.json
│ Project location: ./platformBuilds/harness_ios/RNVApp.xcworkspace
└─────────────────────────────────────────────────────────────────┘
 BUNDLE  ./index.js

 LOG  Running "RNVApp" with {"rootTag":1,"initialProps":{}}
 LOG  FILE WRITTEN!
 LOG  onDeviceChange PORTRAIT
```

**Comment by RicardasN** [comment]
Also checked the issue and connot reproduce. Tested on `release/1.0` branch when the latest release was `1.0.0-rc.19`
